File: src/python/day1/input.py
import os

def readInput(text):
    a1 = []
    a2 = []
    for line in text:
        nums = line.split()
        a1.append(int(nums[0]))
        a2.append(int(nums[1]))
    return a1, a2

def readFile(file):
    with open(file) as f:
        return readInput(f.readlines())

a1, a2 = readFile(os.path.join(os.path.dirname(__file__), "input.txt"))

if __name__ == "__main__":
    print(a1, a2)

File: src/go/day21/input.go
package main

import (
	"bufio"
	"bytes"
	_ "embed"
	"io"
	"regexp"
	"strconv"

	"github.com/dancantos/advent2024/src/go/pkg/must"
)

//go:embed input.txt
var input []byte

var codes, nums = readInput(bytes.NewReader(input))

var numRegex = regexp.MustCompile(`\d+`)

func readInput(r io.Reader) ([]string, []int) {
	codes := []string{}
	nums := []int{}
	s := bufio.NewScanner(r)
	s.Split(bufio.ScanLines)
	for s.Scan() {
		codes = append(codes, s.Text())
		n := numRegex.FindString(s.Text())
		nums = append(nums, must.Return(strconv.Atoi(n)))
	}
	return codes, nums
}
